### PR TITLE
[Vue] migrate remaining directives in CorePluginsAdmin

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18432.**

Changes:
* Migrate piwik-save-button.
* Migrate piwik-form.
* Migrate piwik-plugin-filter.
* Migrate piwik-plugin-management.
* Migrate piwik-plugin-upload.
* Migrate piwik-save-button.
* Allow event names to be different in vue than in angularjs.
* Support '&?' scope binding type in createAngularJsAdapter.

Changes made to make UI tests pass:
* Adding a polyfill for AbortController (since hacking an .abort() method on a promise isn't good enough). The polyfill is recommended by github's `fetch()` polyfill library: https://github.com/github/fetch. See the AbortController spec: https://developer.mozilla.org/en-US/docs/Web/API/AbortController

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests should be passing for this one

**Review by peterhashair — APPROVED** [review]

**Review by peterhashair — APPROVED** [review]
probably wait to if there is any other comments

File: plugins/CorePluginsAdmin/angularjs/form/form.directive.js [hunk 1]
```diff
@@ -25,15 +25,9 @@
 
                     $timeout(function () {
```
Comment (sgiehl) [inline plugins/CorePluginsAdmin/angularjs/form/form.directive.js]: Any reason why this file is still needed?

**Review by sgiehl — APPROVED** [review]
Code overall looks fine. Did a quick testing of the UI and it seems everything works fine.
The JS tests are failing, but that seems to be caused by the last PR already. Might be good to fix them before merging.
UI test failures seem to be unrelated to this PR.

**Pull request merged**